### PR TITLE
[FTAMS102020] Rich Alossi

### PR DESCRIPTION
**Pull request by alossix**

### CODE REVIEW

**Comment by alossix** [comment]
Misnamed last commit - Iterations 1-4 done; did not complete Iteration 5.

File: js/index.js [hunk 1]
```diff
@@ -1,42 +1,49 @@
 // ITERATION 1
-
 function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
-
-  //... your code goes here
+  let price = product.querySelector(".price span");
```
Comment (wax-bear) [inline js/index.js:3]: Will any of the variables in this function be reassigned? If not, would "const" be a better variable?

File: js/index.js [hunk 2]
```diff
@@ -1,42 +1,49 @@
...
+  let quantity = product.querySelector(".quantity");
+  let priceVal = price.innerHTML;
+  let quantityVal = product.querySelector("input").value;
```
Comment (wax-bear) [inline js/index.js:6]: Could lines 4 and 6 be combined in a way?

File: js/index.js [hunk 3]
```diff
@@ -1,42 +1,49 @@
...
+  let subtotalVal = priceVal * quantityVal;
+  let subtotal = product.querySelector(".subtotal span");
```
Comment (wax-bear) [inline js/index.js:8]: Could the name of this variable be more descriptive/corresponding to what it describes? For example "subtotalSpanElement".

File: js/index.js [hunk 4]
```diff
@@ -1,42 +1,49 @@
...
+  document.querySelector("#total-value span").innerText = totalPrice;
 }
-
 // ITERATION 4
-
 function removeProduct(event) {
   const target = event.currentTarget;
-  console.log('The target in remove is:', target);
-  //... your code goes here
+  console.log("The target in remove is:", target);
```
Comment (wax-bear) [inline js/index.js:25]: Is this console.log still valuable?

File: js/index.js [hunk 5]
```diff
@@ -1,42 +1,49 @@
...
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
-
-  // ITERATION 2
-  //... your code goes here
-
+  let totalPrice = 0;
+  let products = document.querySelectorAll(".product");
```
Comment (wax-bear) [inline js/index.js:15]: Will products be reassigned?

File: js/index.js [hunk 6]
```diff
@@ -1,42 +1,49 @@
...
+  calculateAll();
 }
 
 // ITERATION 5
 
-function createProduct() {
-  //... your code goes here
+function createProduct(productName, priceVal) {
+  let newProd = document.createElement("tr");
+  let productName = newProd.classList.add("product");
```
Comment (wax-bear) [inline js/index.js:34]: I believe this variable declaration causes an error, as productName is already declared as an argument.

File: js/index.js [hunk 7]
```diff
@@ -1,42 +1,49 @@
...
+  newProd.innerHTML = `<td class="name"><span></td>`;
```
Comment (wax-bear) [inline js/index.js:35]: Is the HTML you are assigning actually the HTML you want to assign to nhe newProd.innerHTML?

File: js/index.js [hunk 8]
```diff
@@ -1,42 +1,49 @@
...
+
+  const removeBtn = document.querySelectorAll(".btn-remove");
+  removeBtn.forEach((btn) => {
+    btn.addEventListener("click", removeProduct);
+  });
 
-  //... your code goes here
+  const createBtn = document.getElementById("create");
+  createBtn.addEventListener("click", createProduct);
 });
```
Comment (wax-bear) [inline js/index.js:49]: Good stuff Rich!
Most of the functionalities in the UI work. However, I cannot create a new product.
Please check out my comments for potential improvements.

**Pull request closed without merging**